sources/lib/api/gcp/stackdriver/monitoring/_impl/api.py

```diff
@@ -0,0 +1,283 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""HTTP API wrapper for the Google Monitoring API."""
+
+try:
+  from gcp._util import Http, RequestException
+except ImportError:
+  # Use a minimal substitute module if the Datalab library is not available.
+  from .http import Http, RequestException
```

Comment:
When the base library (contents of _impl) is being used as a standalone
module in a context other than gcp.stackdriver.monitoring. We didn't want
to maintain two versions of this code, one that depends on gcp._util, and
one that doesn't.

On Fri, Apr 8, 2016 at 7:40 PM, Graham Wheeler <notifications@github.com>
wrote:

> In sources/lib/api/gcp/stackdriver/monitoring/_impl/api.py
> <https://github.com/GoogleCloudPlatform/datalab/pull/795#discussion_r59103131>
> :
>
> > +#
> > +# http://www.apache.org/licenses/LICENSE-2.0
> > +#
> > +# Unless required by applicable law or agreed to in writing, software
> > +# distributed under the License is distributed on an "AS IS" BASIS,
> > +# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +# See the License for the specific language governing permissions and
> > +# limitations under the License.
> > +
> > +"""HTTP API wrapper for the Google Monitoring API."""
> > +
> > +try:
> > +  from gcp._util import Http, RequestException
> > +except ImportError:
> > +  # Use a minimal substitute module if the Datalab library is not available.
> > +  from .http import Http, RequestException
>
> Under what circumstances do you think the Datalab library would not be
> available?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/datalab/pull/795/files/02f77850899536b23933224d882e31b426170b7d#r59103131>
>